chainerx_cc/chainerx/routines/creation.cc

```diff
@@ -345,4 +347,60 @@ Array Linspace(
     return out;
 }
 
+std::vector<Array> Meshgrid(const std::vector<Array>& arrays, MeshgridIndexingMode mode) {
+    Shape shape;
+    Shape broadcast_shape;
+    std::vector<Shape> arr_reshapes;
+    std::vector<Array> grid_arrays;
+    grid_arrays.reserve(arrays.size());
+
+    // Algo
+    //
+    // Step 1: Reshape/View each array as broadcastable based
+    // on number of input vectors.
+    // Eg. For tuple of vectors (n1, n2, n3)
+    // where ni is length of that vector.
+    // After this step for Vector 1 , we will reshape it as
+    // (n1, 1, 1) , Vector 2 as (1, n2, 1)
+    //
+    // Step 2: Broadcast each vector to the shape
+    // if (indexing == "ij") -> (n1, n2, n3)
+    // else if (indexing == "xy") -> (n2, n1, n3)
+    // Note : For "xy" only n1 and n2 swap their places
+    //        all others are same as "ij"
+
+    // Step 1
+    for (const Array& array : arrays) {
+        shape.emplace_back(1);
+        broadcast_shape.emplace_back(array.GetTotalSize());
+    }
+
+    // Shape for each array based on number of arrays.
+    for (size_t i = 0; i < arrays.size(); ++i) {
+        Shape temp_shape(shape.begin(), shape.end());
+        temp_shape[i] = arrays[i].GetTotalSize();
+        arr_reshapes.push_back(temp_shape);
```

Comment:
This is also a detail and you may or may not skip this, but we almost always prefer `emplace_back` over `push_back` since there are basically not pros with the latter and the latter falls back to the former.

Reply:
Oh Thank you. Fixed.